Bottom: 1e4efc04dbb9c3d452cda0a04f348309fb9ce8dc
Top:    f1eb866ee7250c40fa683813781207364a8f0331
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-13 20:34:10 +0100

Try optimising polygon operations by keeping an rtree with their contours

InsertHoles() already creates an rtree of contours to do its internal
processing, and it is possible that intersect() could be sped up by
its existance as well.




---

diff --git a/src/polyarea.h b/src/polyarea.h
index 42acc1a..819b074 100644
--- a/src/polyarea.h
+++ b/src/polyarea.h
@@ -128,6 +128,7 @@ struct POLYAREA
 {
     POLYAREA *f, *b;
     PLINE *contours;
+    void *contour_tree;
 };
 
 BOOLp poly_M_Copy0(POLYAREA ** dst, const POLYAREA * srcfst);
